Events.md

```diff
@@ -0,0 +1,225 @@
+
+If the posthook of the offer fails. Then we emit `OfferFailWithPosthookData` instead of just `OfferFail`. This event has one extra field, which is the reason for the posthook failure. By emitting the posthook data, an indexer can keep track of the reason posthook fails, this could for example be used for analytics.
+
+This event is emitted doring posthook end, we wait to emit this event to the end, because we need the information of `penalty`, which is only available at the end of the posthook. This means that `OfferFail` events are emitted in reverse order, compared to what order they are taken. This is do to the way we handle posthooks. The same goes for `OfferSuccess`.
+
+By emitting this event, an indexer can keep track of, if an offer failed and thereby if the offer is live. By emitting the wants and gives that the offer was taken with, then an indexer can keep track of these amounts, which could be useful for e.g. strategy manager, to know if their offers fail at a certain amount.
+
+### OfferSuccess
+
+This event is emitted when an offer is successfully taken. Meaning both maker and taker has gotten their funds.
+
+It emits the `offerlist key`, `taker` address, `maker` address, the `offerId` and the `takerWants` and `takerGives` that the offer was taken at. Just as for `OfferFail`, the `offerlist key`, `taker` and `maker` are not needed for an indexer to work, but are needed for RPC calls. So they are emitted and indexed for that reason. Just like `OfferFail` this event is emitted during posthook end, so it is emitted in reverse order compared to what order they are taken. This event could be emitted at offer execution, but for consistency we emit it at posthook end.
+
+If the posthook of the offer fails. Then we emit `OfferSuccessWithPosthookData` instead of just `OfferSuccess`. This event has one extra field, which is the reason for the posthook failure. By emitting the posthook data, an indexer can keep track of the reason posthook fails, this could for example be used for analytics.
+
+By emitting `offerId`, `wants` and `gives`, an indexer can keep track of whether the offer was partially or fully taken, by looking at the what the offer was posted at.
+
+### OrderComplete
+
+This event is emitted when a market order is finished.
+
+It only emits the total `fee paid`. We need this event, in order to know that a market order is completed. This way an indexer can know exactly in what context we are in.
+The total `fee paid` for that market order, is needed, as we do not emit this any other place. It is not indexed as there is no need for RPC calls to filter on this.
```

Comment:
Perhaps this event should have the same indexed fields as `OrderStart` so RPC clients can filter on this and get the complete picture for the order?

Reply:
Might not be a bad idea. @adhusson @lnist   WDYT?

Reply:
I agree

Reply:
I've implemented this now.